src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RetriesSpec.groovy

```diff
@@ -446,4 +448,170 @@ and at least 1 path must remain safe"
+        given: "A flow, with src switch supporting multi-table (currently in single-table)"
+        def swPair = topologyHelper.switchPairs.find {
+            it.src.features.contains(SwitchFeature.MULTI_TABLE) && it.paths.size() > 1 &&
+                    it.paths.find { it.size() > 2 }
+        }
+        assumeTrue("Couldn't find a switch pair with src sw supporting Multi-table, 2+ paths and at least 1 " +
+                "path having transit switch", swPair as boolean)
+        def originalMode = swPair.src.changeMultitable(false)
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+        print flow.flowId
+        def path = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+
+        and: "Only 1 alt path is available, that path has a transit switch"
+        def otherPath = swPair.paths.find { it != path && it.size() > 2 }
+        List<Isl> islsToKill = []
+        def involvedIsls = pathHelper.getInvolvedIsls(path) + pathHelper.getInvolvedIsls(otherPath)
+        swPair.paths.findAll { it != path && it != otherPath }.each {
+            pathHelper.getInvolvedIsls(it).findAll { !(it in involvedIsls || it.reversed in involvedIsls) }.each {
+                islsToKill.add(it)
+            }
+        }
+        def broughtDownIsls = islsToKill.unique { a, b -> a == b || a == b.reversed ? 0 : 1 }
+        broughtDownIsls.every { antiflap.portDown(it.srcSwitch.dpId, it.srcPort) }
+        wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == islsToKill.size() * 2
+        }
+
+        when: "Change src switch mode to multi-table"
+        swPair.src.changeMultitable(true)
+
+        and: "Transit switch on alt path is false-active, so that on reroute system fails to install rules and rollback"
+        def currentSwitches = pathHelper.getInvolvedSwitches(path)
+        def otherSwitches = pathHelper.getInvolvedSwitches(otherPath)
+        def brokenSwitch = otherSwitches.find { !currentSwitches.contains(it) }
+        def blockData = switchHelper.knockoutSwitch(brokenSwitch, RW)
+        def swDown = true
+        database.setSwitchStatus(brokenSwitch.dpId, SwitchStatus.ACTIVE)
+
+        and: "Break ISL on current path to init a reroute"
+        def otherIsls = pathHelper.getInvolvedIsls(otherPath)
+        def isl = pathHelper.getInvolvedIsls(path).find { !otherIsls.contains(it) }
+        antiflap.portDown(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "At least 2 reroute attempts happen, but more retries are about to go"
+        wait(WAIT_OFFSET * 2) {
+            //wait for 2, max is 4
+            assert northbound.getFlowHistory(flow.flowId).count { it.action == REROUTE_ACTION } == 2
+            northboundV2.getFlowStatus(flow.flowId).status == FlowState.DOWN
+        }
+
+        when: "Switch one the new path becomes responsive and allows rule installation"
```

Comment:
The same as in the previous test
Shouldn't we do it in a legal way?
I mean, set the real status for the switch and then invoke the `reviveSwitch` method

Reply:
I can agree with the previous situation where we do this in cleanup, but right here the whole situation is kinda illegal. We simulate the case when switch is active but we can't install rules on it - this is already illegal. And right here we simulate that one of the retries suddenly succeed. Switch should not officially go 'offline' in this particular scenario. It just became responsive again in the middle of retries. Making it officially offline may ruin the whole point of test